project_generator/project.py

```diff
@@ -197,21 +193,37 @@ def __init__(self, name, project_files, pgen_workspace):
                raise IOError("The file %s referenced in main yaml doesn't exist."%project_file)
 
     def _fill_project_defaults(self):
-        self.project['name'] = self.name
-        self.project['output_type'] = self.output_types['executable']
-        self.project['linker_file'] = None
-        self.project['project_name'] = None
-        self.project['tools_supported'] = [self.workspace.settings.DEFAULT_TOOL]
-        self.project['debugger'] = 'cmsis-dap'
-        self.project['build_dir'] = 'build'
-        self.project['output_dir'] = {
-            'path': '',
-            'rel_path': '',
-            'rel_count': '',
+
+        self.project = {
+            'name': self.name,          # project name
+            'core': '',                 # core
+            'linker_file': None,          # linker command file
+            'build_dir' : 'build',
+            'debugger' : 'cmsis-dap',   # TODO: find out what debugger is connected
+            'includes': [],             # include paths
+            'source_paths': [],         # source paths
+            'source_files_c': [],       # c source files
+            'source_files_cpp': [],     # c++ source files
+            'source_files_s': [],       # assembly source files
+            'source_files_obj': [{}],   # object files
```

Comment:
Shouldn't be that way once flushed out in customize function. But, this is what it looked like in Valinor.